edgelet/iotedge/src/check/checks/proxy_settings.rs

```diff
@@ -0,0 +1,55 @@
+use failure::{self, Context};
+
+use crate::check::{Check, CheckResult, Checker, CheckerMeta};
+
+#[derive(Default, serde_derive::Serialize)]
+pub(crate) struct ProxySettings {}
+
+#[async_trait::async_trait]
+impl Checker for ProxySettings {
+    fn meta(&self) -> CheckerMeta {
+        CheckerMeta {
+            id: "proxy-settings",
+            description: "proxy settings are consistent in iotedged, moby daemon and config.toml",
+        }
+    }
+
+    async fn execute(&mut self, check: &mut Check) -> CheckResult {
+        self.inner_execute(check)
+            .await
+            .unwrap_or_else(CheckResult::Failed)
+    }
+}
+
+impl ProxySettings{
+    async fn inner_execute(&mut self, check: &mut Check) -> Result<CheckResult, failure::Error> {
+        let settings = if let Some(settings) = &mut check.settings {
+            settings
+        } else {
+            return Ok(CheckResult::Skipped);
+        };
+
+        // Pull the proxy address from the aziot-edged settings
+        // for Edge Agent's environment variables.
+        let edge_agent_proxy_uri = settings.base.agent.env().get("https_proxy");
+
+        // Pull the proxy address from the evironment for Moby and Edge Daemon
+        let moby_proxy_uri = std::env::var("HTTPS_PROXY")
+                                .ok();
+        let edge_daemon_proxy_uri = std::env::var("https_proxy")
+                                .ok();
+
```

Comment:
I believe you'll also want to check the proxy setting for the aziot-identityd daemon as well. You should be able to check it the same way you check aziot-edged.

Reply:
the identity service didn't have the proxy setup. Would this be a difference between runtime version 1.1 and 1.2? My test setup is with 1.2.

Reply:
In 1.1 there was no identity service so it's new for 1.2. My comment was based on the following doc, which for 1.2 mentions having to configure the identity service daemon with the `https_proxy` environment variable:

https://docs.microsoft.com/en-us/azure/iot-edge/how-to-configure-proxy-support?view=iotedge-2020-11&preserve-view=true#linux